### PR TITLE
Config files restructure

### PR DESCRIPTION
**Pull request by swalladge**
The changes here mostly deal with restructuring how/where confile files, scripts, maps, etc. are loaded from. 
- By default, everything is expected to be under `~/.pysnip` (the location can be changed through program arguments)
- Everything in the configs directory follows a neat structure (as detailed in `configs/README.md`)
- I've added a requirements.txt which can be used for installing pip dependencies easily - something could be added to the readme about that (more useful after adding a working setup.py - as in #16 which may need modification after these refactorings)
- A todo: information about config structure should be added to the readme when finalized how it all should work. A script could also be made to copy example config to `~/.pysnip` for getting started.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,32 @@
-![PySnip](http://i.imgur.com/QFgqcRM.png)
 
-[![Build Status](https://travis-ci.org/NateShoffner/PySnip.svg?branch=master)](https://travis-ci.org/NateShoffner/PySnip)
+# About this fork #
```
Comment (swalladge) [inline README.md]: Don't merge in the 'about this fork' part of the readme.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+This fork (original project at [NateSchoffner/PySnip](https://github.com/NateShoffner/PySnip)) has been created with the aim to complete the following:
+
+- [DONE] separate server from configs/maps/scripts/etc (at the moment it's all combined, and messy)
+- make it installable via pip or a linux package manager
+- port to python3
+- keep up to date and fix any bugs (please send pull requests!)
+- provide some more documentation on how to instal, run, and configure the server
+
+
+## FAQs
```
Comment (swalladge) [inline README.md]: This note is a useful one to keep around - whether you want to merge it in with a FAQ in your readme, or start a wiki?

**Comment by noway421** [comment]
This is a great and huge change. I don't really remember the project structure too well to judge whether it correlates with the potential `pysnip-as-a-module` PR or not.

One concern I have is does it break potentially any userscripts? Feels like `feature_server` might be specifically expected to be present by certain scripts..

**Comment by swalladge** [comment]
It's been a while since I looked at this so I could be wrong, but from memory this doesn't change anything in regards to installing as a package. I think once I got the config file parts working I did look into it but got stuck - `enet` is included with it and I couldn't get it to run properly with the enet from upstream (i was attempting to split it out). I dare say it could be included fine with it to be distributed as a package - I'll have to attack it again later.

@noway421 what exactly do you mean by `feature_server` expected to be present? All of that _should_ be still in the same place. 

Also I'd be interested to know if anyone with commit privileges would be willing to review and merge (or give feedback on) these changes. I'll also fix the readme changes that aren't relevant to this.

**Comment by noway421** [comment]
Yep, most probably it is not related to `pysnip-as-a-module`. So in that terms it is something that won't interfere with it. I'd say it is good to merge. Hopefully Nate will take a look into this PR and approve.

I remember I spot some issue with enet too, a better way would be to include it altogether, at least for a first release. I'm seeing `no-enet` branch in your https://github.com/swalladge/PySnip repo. I'm assuming this is the change you attempted. Good job on that.

I feel like some user scripts (which are in `feature_server/scripts/`) might expect `config.txt` and `maps/*.txtc` in their parent directory i.e. in `feature_server`. They'll go one directory up from the `__file__ ` and look for them. To be bullet-proof, and while it is not tested by wide audience, for now it would make sense to make symlinks for `feature_server/data/`, `feature_server/config.txt` and `feature_server/maps/*.txtc`. Although the number of scripts which rely on this behaviour is not great most likely. I can only think about the free build script which allows you to save the map with a command sent by a player.

I'm planning to take a deeper look into pip module and I'll try to base off my changes from your config changes (i.e. from https://github.com/swalladge/PySnip `master`)

**Comment by swalladge** [comment]
I think another good option would be to inspect all the scripts we currently have and update any references to suit the new config layout. I won't have much time over the next few weeks, but after that I'll check it out.

**Comment by noway421** [comment]
I went through your changes again, actually my concerns are unfounded: the userscripts are still relatively the same to maps and config files. They are both in `configs` while initially I thought they stayed in `feature_server` with other pyspades core scripts (by those I mean commands.py and such).

In that case, even if some problems do arise it is possible to fix them within the pysnip repo.

👍  I'd say it's good to merge

**Comment by StripedMonkey** [comment]
As a legitimate thing who approves the merges and stuff? This has been sitting here for quite a while, I'm looking into OpenSpades development and am looking at this and the client and am rather dissapointed in how lackluster the activity is.

**Comment by samuelallan72** [comment]
@StripedMonkey development on PySnip hasn't happened for a long time unfortunately. You might want to check out an active fork that includes this PR and many new improvements: https://github.com/piqueserver/piqueserver/ :)

**Comment by StripedMonkey** [comment]
Cool, I suppose part of the issue is that this is the server management that is advertised on https://www.buildandshoot.com/download/ I didn't see the other one.

**Comment by noway** [comment]
BNS link should really be updated tbh

**Comment by samuelallan72** [comment]
@StripedMonkey @noway see pr --> https://github.com/BuildandShoot/buildandshoot.github.io/pull/12 feedback welcome! Hopefully a maintainer there will be willing to merge. :smile:

Edit: though this isn't the place to discuss this - jump on one of the bridged piqueserver chat channels!

**Comment by samuelallan72** [comment]
Closing this pull request since everything useful in here has been merged to [piqueserver](https://github.com/piqueserver/piqueserver/) and I have no intention to work on or maintain anything in this PR/repo in the future.

**Comment by NotAFile** [comment]
RIP pull-request, you served us well

**Pull request closed without merging**